src/lib.rs

```diff
@@ -317,6 +317,12 @@ When your transaction is complete, enter '\commit' or '\abort' as appropriate."#
     }
 
     async fn handle_autocommit_partiql(&mut self, line: &str) -> Result<()> {
+        if self.deps.opt.auto_commit == AutoCommitMode::Off {
+            // We're not in auto-commit mode, but there is no transaction
+            return Err(QldbShellError::UsageError(format!(
+                "No active transaction and not in auto-commit mode. \
+                Start a transaction with '\\start transaction'")))?
```

Comment:
This works, but `?` is basically a return statement, so it's somewhat redundant to write `return ... ?`.

`?` is the 'try' operator that boils down to:

1. check if the Result is an Err
2. if so, call `.into()`, relying on type inference to figure out the method's return signature, then return it

This is not hardcoded to the `Result` type, rather through infrastructure called the Carrier trait, but for now thinking about it in terms of

```
if let Err(e) = thingy { return e.into() }
```

is a good mental model.

In terms of type inference.. the return type is an `anyhow::Result` which is a typealias for `Result<T, Error>`, where `Error` is a struct defined by the anyhow crate. There is a bunch of machinery provided by the thiserror crate which lands up generating an impl that lets QldbShellError become an Error!

Or, put another way..

1. either remove `return`
2. or remove `?` and call `into` yourself.